feat(messaging): support for `frameId`

Mehdi-Hp opened the pull request:
Adding support for optional `frameId` in `Browser.tabs.sendMessage`

aklinker1 commented on packages/messaging/src/extension.ts:
```diff
@@ -51,9 +60,25 @@ export function defineExtensionMessaging<
+      // Handle both number and options object
+      const options: SendMessageOptions = typeof arg === 'number' ? { tabId: arg } : arg;
+
+      if (typeof options.tabId !== 'number') {
+        throw new Error('tabId is required when sending a message to a tab');
+      }
```
I'd rather just let the browser throw an error if the tab ID is missing or the wrong type.

```suggestion
```

Mehdi-Hp replied:
Fair enough. Done.

aklinker1 reviewed (changes requested):
Thanks for the PR and updating the tests.

One small change, I prefer to avoid runtime type-checks if typescript already reports as fine. The error the browser throws in those cases is good enough.

aklinker1 commented:
Looks like there are some type errors in `src/extension.test-d.ts`

Mehdi-Hp commented:
> Looks like there are some type errors in `src/extension.test-d.ts` causing checks to fail

Fixed.
Updated the docs too, but I think that needs your edit.
